### PR TITLE
Testing framework and Servicelayer

### PR DESCRIPTION
**Pull request by epichub**
This PR contains integration testing and the start of a service layer for concept creation.

### CODE REVIEW

File: src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java [hunk 1]
```diff
@@ -91,6 +91,38 @@ public Query getSimilarCasesByID(
             @ApiResponse(code = 500, message = "Failure")
     })
     public HashMap<String, HashMap<String,Double>> getSimilarInstancesByIDWithinIDs(
+            @PathVariable(value="conceptID") String conceptID,
+            @PathVariable(value="casebaseID") String casebaseIDs,
```
Comment (kerstinbach) [inline src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java]: casebaseIDs should be casebaseID as there is only one.

File: src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java [hunk 2]
```diff
@@ -91,6 +91,38 @@ public Query getSimilarCasesByID(
...
+            @RequestParam(value="caseID") String caseID,
+            @RequestParam(value="querybaseIDset") String querybaseIDsJson,
```
[the remark this answers is not in the document]
Comment (epichub) [reply src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java]: I agree with all points - What should we call the querybaseIDs? casebasesubsetIDs ?

File: src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java (hunk 1 above)
Comment (epichub) [reply src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java]: Agree.
I will submit a new commit when we resolve the parameter naming.

File: src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (kerstinbach) [reply src/main/java/no/ntnu/mycbr/rest/controller/RetrievalController.java]: I'm not sure about the naming, but subsetCaseIDs or filterCaseIDs can be options.

**Comment by epichub** [comment]
All comments should be fixed in commit 26a355d613f33e634257015c862c17f3cc72efab .

**Pull request merged**